presto-spi/src/main/java/com/facebook/presto/spi/relation/UnresolvedSymbolExpression.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.relation;
+
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.spi.SourceLocation;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.lang.String.format;
+import static java.util.Collections.emptyList;
+import static java.util.Objects.requireNonNull;
+
+public class UnresolvedSymbolExpression
```

Comment:
SubqueryPlanner does it. It doesn't explicitly check for UnresolvedSymbolExpression, but has its own way to identify which expressions cannot be resolved. 

This isn't very clean because in planner we use a mix of expressions and rowexpressions.